### PR TITLE
Tests/new core param val data

### PR DESCRIPTION
**Pull request by neilSchroeder**
## Summary of changes and related issue
This pull request adds a comprehensive unit test suite for the `DataValidator` class in the new core parameter validation module. The tests cover initialization, query validation logic, and edge cases related to the use of the "localize" processor for different datasets and variables.

Unit test coverage enhancements:

* Added a new test class `TestDataValidator` in `tests/new_core/param_validation/test_data_param_validator.py` with setup and multiple test cases to validate the behavior of `DataValidator`, including catalog key initialization and catalog assignment.
* Implemented tests for the `_check_query_for_wrf_and_localize` method to verify correct handling of queries with/without the localize processor, WRF activity, and variable checks, including proper warning emission for invalid configurations.
* Added tests for the `is_valid_query` method to ensure it returns `None` and emits warnings on invalid localize usage, and correctly delegates to the parent method when checks pass.

## Relevant motivation and context
tests are good

## How to test 
ci tests and `python -m pytest tests/new_core/param_validation/test_data_param_validator.py --cov=climakitae/new_core/param_validation/`

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] New tests (increased test coverage on existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] None of the above  

## To-Do
- [x] Unit tests
  - [ ] Existing unit tests are passing
  - [ ] If relevant, new unit tests are written (required 80% unit test coverage)
  - [ ] Advanced Testing passing (select Advanced Testing label)
- [x] Documentation
  - [ ] Intent of all functions included
  - [ ] Complex code commented
  - [ ] Functions include [NumPy style docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html) 
- [x] Naming conventions followed
  - [ ] Helper functions hidden with `_` before the name
- [x] Any notebooks known to utilize the affected functions are still working
- [x] Black formatting has been utilized
- [x] Tagged/notified 2 reviewers for this PR

### CODE REVIEW

File: tests/new_core/param_validation/test_data_param_validator.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            "activity_id": "LOCA2",  # Not WRF, should fail
+            "variable_id": "tas",
+        }
+
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            result = self.validator.is_valid_query(query)
+
+            assert result is None
+            assert len(w) == 1
```
Comment (claalmve) [inline tests/new_core/param_validation/test_data_param_validator.py:73]: I know that this is out of the scope of this PR, but why does this `is_valid_query` call return a list instead of just None?

Comment (neilSchroeder) [reply tests/new_core/param_validation/test_data_param_validator.py:73]: I'm confused about your question.

this test shows asserts that 
1. `is_valid_query()` MUST be `None` for this test
2. the number of errors (`len(w)`) should be 1. 

`is_valid_query()` should never return a list
